src/main/resources/db/migration/V1.0.0__initial_schema.sql

```diff
@@ -0,0 +1,451 @@
+    mark integer check ((mark >= 1) and (mark <= 5)),
+    count integer not null check (count > 0),
+    primary key (subject_id, mark)
+);
+
+create table subject_tag
+(
+    id   uuid primary key,
+    name varchar(128) not null,
+    subjects_count integer not null check (subjects_count > 0)
+);
+
+create table subject_to_tag
+(
+    subject_id uuid references subject,
+    tag_id     uuid references subject_tag on delete cascade,
+    primary key (subject_id, tag_id)
+);
+
+create table review
+(
+    id             uuid primary key,
+    title          varchar(128),
+    reviewer_id    uuid not null references actor,
+    subject_id     uuid not null references subject,
+    mark           integer not null check (mark >= 1 and mark <= 5),
+    is_shown       boolean not null,
+    upvotes_count   integer not null check (upvotes_count >= 0),
+    downvotes_count integer not null check (downvotes_count >= 0)
+);
+
+create table review_body
+(
+    id                uuid primary key,
+    review_id         uuid not null references review,
+    content           varchar not null,
+    --TODO: last_modified_timestamp timestamp,
+    created_timestamp timestamp not null
+);
+--TODO?: index by review_body_created_timestamp
+
+
+create type review_point_type as enum ('ADVANTAGE', 'DISADVANTAGE');
+
+create table review_point
+(
+    review_id uuid references review,
+    ordering  int not null check (ordering >= 0),
+    type      review_point_type not null,
+    content   varchar(256) not null,
+    primary key (review_id, ordering)
+);
+-- no sense in creating index by ordering because it does not guarantee select order
+
+create table review_image
+(
+    id        uuid primary key,
+    review_id uuid not null references review,
+    ordering  integer not null check (ordering >= 0)
+);
+-- no sense in creating index by ordering because it does not guarantee select order
+
+create type vote_type as enum ('UP', 'DOWN');
+
+create table review_vote
+(
+    review_id   uuid references review,
+    reviewer_id uuid references actor,
+    type        vote_type not null,
+    primary key (review_id, reviewer_id)
+);
+
+create table review_comment
+(
+    id                uuid primary key,
+    review_id         uuid not null references review,
+    author_id         uuid not null references actor,
+    content           varchar not null,
+    created_timestamp timestamp not null,
+    --TODO: last_modified_timestamp timestamp,
+    is_shown          boolean not null,
+    upvotes_count      integer not null check (upvotes_count >= 0),
+    downvotes_count    integer not null check (downvotes_count >= 0)
+);
+
+create table review_comment_vote
+(
+    review_comment_id uuid references review_comment,
+    reviewer_id       uuid references actor,
+    type              vote_type not null,
+    primary key (review_comment_id, reviewer_id)
+);
+
+create type report_issuer_type as enum ('GUEST', 'REVIEWER', 'BRAND_OWNER', 'SYSTEM');
```

Comment:
I've changed how we store our actors and now we don't need this enum

Reply:
removed